flytepropeller/pkg/controller/garbage_collector.go

```diff
@@ -162,7 +162,7 @@ func (g *GarbageCollector) StartGC(ctx context.Context) error {
 		logger.Warningf(ctx, "Garbage collector is disabled, as ttl [%d] is <=0", g.ttlHours)
 		return nil
 	}
-	ticker := g.clk.NewTicker(g.interval)
+	ticker := g.clk.NewTimer(g.interval)
```

Comment:
Why was this changed from a ticker to a timer? I'm a total golang noob but I think this means that flytepropeller will only garbage collect once each time it restarts. 

I think this is causing us some problems https://flyte-org.slack.com/archives/CP2HDHKE1/p1701118219768919?thread_ts=1700590318.336129&cid=CP2HDHKE1